pdns/dnsdist-lua2.cc

```diff
@@ -769,6 +769,59 @@ void moreLua(bool client)
 #endif
       });
 
+
+// --------------------------------------------------------------------------
+// GCA - Seth Ornstein added lua callable functions - 6/2/2017
+// DNSDistProtoBufMessage - setTag, setTagArray, setProtobufResponseType, setProtobufResponseTypeQT
+
+     g_lua.registerFunction<void(DNSDistProtoBufMessage::*)(std::string, std::string)>("setTag", [](DNSDistProtoBufMessage& message, const std::string& strLabel, const std::string& strValue) {
+
+      setLuaSideEffect();
+
+      message.addTags(strLabel, strValue);               // add a tag to store text - not used by dnsdist normally
+     });
+
+
+     g_lua.registerFunction<void(DNSDistProtoBufMessage::*)(vector<pair<string, string>>)>("setTagArray", [](DNSDistProtoBufMessage& message, const vector<pair<string, string>>&tags) {
+
+      setLuaSideEffect();
+
+      for (const auto& tag : tags)
+        {
+          message.addTags(tag.first, tag.second);               // add a tag to store text - not used by dnsdist normally
+        }
+     });
+
+
+                                                                // setProtobufResponseType - no timestamp
+     g_lua.registerFunction<void(DNSDistProtoBufMessage::*)(const std::string&)>("setProtobufResponseType", [](DNSDistProtoBufMessage& message, const std::string& strQueryName) {
+
+        message.setType(DNSProtoBufMessage::Response);          // set protobuf type to be response - not query
+
+        message.setQueryTime(0, 0);                             // seconds and microseconds
+
+        message.addRRs(strQueryName);                           // add a RR to the response
```

Comment:
New function setProtobufResponseRR() added.  setProtobufResponseType() does not add a RR.